From 50bca87976b8a8a4be094c26e4fba82d2fe3ba51 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 9 Oct 2015 11:16:40 -0700
Subject: [PATCH 24/26] net: mdio-octeon: Quit using readq_relaxed/writeq_relaxed
Patch-mainline: Not yet, Cavium has to do so
References: fate#319484

On early versions of the ThunderX SoCs, readq_relaxed/writeq_relaxed
had the property of ensuring that reads and writes were properly
ordered, but were higher performance that readq/writeq.  The ordering
is no longer guaranteed, and since MDIO buses do not require the very
highest performance register access, the safest thing to do is just
use normal readq/writeq.  So, do that.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>

---
 drivers/net/phy/mdio-octeon.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/phy/mdio-octeon.c
+++ b/drivers/net/phy/mdio-octeon.c
@@ -126,8 +126,8 @@ static u64 oct_mdio_readq(u64 addr)
 	return cvmx_read_csr(addr);
 }
 #else
-#define oct_mdio_writeq(val, addr)	writeq_relaxed(val, (void *)addr)
-#define oct_mdio_readq(addr)		readq_relaxed((void *)addr)
+#define oct_mdio_writeq(val, addr)	writeq(val, (void *)addr)
+#define oct_mdio_readq(addr)		readq((void *)addr)
 #endif
 
 static void octeon_mdiobus_set_mode(struct octeon_mdiobus *p,
